src/ports.rs

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use std::sync::Arc;
+
+use anyhow::Context;
+use enarx_config::{Config, File};
+use static_init::dynamic;
+use tokio::sync::{RwLock, RwLockWriteGuard};
+
+#[dynamic]
+static PORTS_IN_USE: Arc<RwLock<Vec<u16>>> = Arc::new(RwLock::new(vec![]));
+
+pub fn get_listen_ports(toml: &str) -> anyhow::Result<Vec<u16>> {
+    let config = toml::from_str::<Config>(toml).with_context(|| "failed to parse enarx config")?;
+
+    let ports = config
+        .files
+        .into_iter()
+        .filter_map(|file| match file {
+            File::Null { .. }
+            | File::Stdin { .. }
+            | File::Stdout { .. }
+            | File::Stderr { .. }
+            | File::Connect { .. } => None,
+            File::Listen { port, .. } => Some(port),
```

Comment:
Let's add a match on protocol here as well, which make sure it's either tcp or tls. We want a compilation error here if we ever add something else (e.g. UDP)